src/mscorlib/src/System/String.cs

```diff
@@ -1208,6 +1272,42 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
             }
             return foundCount;
         }
+
+        private struct SplitSeparatorArray
+        {
+            private readonly bool _single;
+            private readonly string _separator;
+            private readonly string[] _separators;
+
+            public SplitSeparatorArray(string separator)
+            {
+                _single = true;
+                _separator = separator;
+                _separators = null;
+            }
+
+            public SplitSeparatorArray(string[] separators)
+            {
+                _single = false;
+                _separator = null;
+                _separators = separators;
+            }
+
+            public bool IsNullOrEmpty
+            {
+                get { return !_single && (_separators == null || _separators.Length == 0); }
```

Comment:
I worry about the new branches here.  Is it possible to do something similar to the char[] path where we have string* and just stackalloc for the single item case?

It would be good to understand how much slower the existing code got with this this new thing.